profiling/src/lib.rs

```diff
@@ -548,13 +592,50 @@ extern "C" fn rinit(r#type: c_int, module_number: c_int) -> ZendResult {
                 };
                 if let Err((index, interrupt)) = profiler.add_interrupt(interrupt) {
                     warn!(
-                        "VM interrupt {} already exists at offset {}",
-                        index, interrupt
+                        "VM interrupt {index} already exists at offset {interrupt}"
                     );
                 }
             }
         });
     }
+
+    #[cfg(feature = "allocation_profiling")]
+    {
+        if profiling_experimental_allocation_enabled {
+            if !is_zend_mm() {
+                // Neighboring custom memory handlers found
+                trace!("Found another extension using the ZendMM custom handler hook");
+                unsafe {
+                    zend::zend_mm_get_custom_handlers(
+                        zend::zend_mm_get_heap(),
+                        &mut PREV_CUSTOM_MM_ALLOC,
+                        &mut PREV_CUSTOM_MM_FREE,
+                        &mut PREV_CUSTOM_MM_REALLOC,
+                    );
+                }
+            }
+
+            unsafe {
+                zend::ddog_php_prof_zend_mm_set_custom_handlers(
+                    zend::zend_mm_get_heap(),
+                    Some(alloc_profiling_malloc),
+                    Some(alloc_profiling_free),
+                    Some(alloc_profiling_realloc),
+                );
+            }
+
+            if is_zend_mm() {
+                info!("Memory allocation profiling could not be enabled. Please feel free to fill an issue stating the PHP version and installed modules. Most likely the reason is your PHP binary was compiled with `ZEND_MM_CUSTOM` being disabled.");
```

Comment:
I'd change this to `warn`, I think? This only triggers if allocation profiling is enabled and it default to off, so when that combines, I think it's more serious than `info`.